webapp/src/test/java/com/box/l10n/mojito/service/tm/TMTextUnitHistoryServiceTest.java

```diff
@@ -0,0 +1,266 @@
+            } catch (RepositoryLocaleCreationException e) {
+                throw new RuntimeException(e);
+            }
+
+            asset = assetService.createAssetWithContent(repository.getId(), "test-asset-path.xliff", "test asset content");
+
+            //make sure asset and its relationships are loaded
+            asset = assetRepository.findOne(asset.getId());
+
+            assetId = asset.getId();
+            tmId = repository.getTm().getId();
+        }
+    }
+
+    @Test
+    public void testEmptyHistory() throws RepositoryNameAlreadyUsedException {
+        createTestData();
+
+        logger.debug("Done creating data for test, start testing");
+
+        Long addTextUnitAndCheck1 = addTextUnitAndCheck(tmId, assetId, "name", "this is the content", "some comment", "3063c39d3cf8ab69bcabbbc5d7187dc9", "cf8ea6b6848f23345648038bc3abf324");
+
+        Locale frFRLocale = localeService.findByBcp47Tag("fr-FR");
+
+        // then get the history of it without adding any variants
+        List<TMTextUnitVariant> history = tmHistoryService.findHistory(frFRLocale.getId(), addTextUnitAndCheck1);
+        assertNotNull(history);
+        assertTrue(history.isEmpty());
+    }
+
+    @Test
+    public void testHistoryOneVariant() throws RepositoryNameAlreadyUsedException {
+        createTestData();
+
+        logger.debug("Done creating data for test, start testing");
+
+        Long addTextUnitAndCheck1 = addTextUnitAndCheck(tmId, assetId, "name", "this is the content", "some comment", "3063c39d3cf8ab69bcabbbc5d7187dc9", "cf8ea6b6848f23345648038bc3abf324");
+
+        logger.debug("Add a current translation for french");
+        Locale frFRLocale = localeService.findByBcp47Tag("fr-FR");
+
+        logger.debug("TMTextUnit tmTextUnit = tmTextUnitRepository.findByMd5AndTmIdAndAssetId(md5, assetId, tmId);");
+        String md5 = tmService.computeTMTextUnitMD5("name", "this is the content", "some comment");
+        TMTextUnit tmTextUnit = tmTextUnitRepository.findByMd5AndTmIdAndAssetId(md5, tmId, assetId);
+        logger.debug("tmtextunit: {}", tmTextUnit);
+
+        TMTextUnitVariant addCurrentTMTextUnitVariant = addCurrentTMTextUnitVariant(tmTextUnit.getId(), frFRLocale.getId(), "FR[this is the content]", "0a30a359b20fd4095fc17fb586e8db4d");
+
+        // then get the history of it without adding any variants
+        List<TMTextUnitVariant> history = tmHistoryService.findHistory(frFRLocale.getId(), addTextUnitAndCheck1);
+        assertNotNull(history);
+        assertFalse(history.isEmpty());
+
+        Iterator<TMTextUnitVariant> iterator = history.iterator();
+        assertTrue(iterator.hasNext());
+
+        TMTextUnitVariant first = iterator.next();
+
+        assertEquals(first.getContent(), "FR[this is the content]");
+        assertEquals(first.getLocale().getBcp47Tag(), "fr-FR");
+        assertEquals(first.getStatus(), TMTextUnitVariant.Status.APPROVED);
+        assertNotNull(first.getCreatedByUser());
```

Comment:
Make sure there is just one history by checking there is no more result after the first one:
```
assertFalse(iterator.hasNext());
```